tick/solver/scpg.py

```diff
@@ -247,7 +251,7 @@ def set_prox(self, prox: Prox):
         """
         self._set('prox', prox)
         self.prox_ssc.set_original_prox(prox)
-        return self
+        return SolverFirstOrder.set_prox(self, prox)
```

Comment:
Do you call this for checks only? I would call this before `self._set('prox', prox)` to perform checks and then override set_prox behavior as intended